### PR TITLE
[CELEBORN-616] Avoid rpc failure response cause memory leak

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
We meet a case in which worker A triggers pause push data and nearly 200w replicate data requests to worker B failed and revived many of the RpcFailure responses. 

Although worker A trims many times, it's till keeps the same direct memory usage. Here the RpcFailure only cleaned after GC, but this memory usage won't cause GC and  the usage won't release. Keep this worker trim and trim again, won't recover. In this pr, we also release RPC failure responses's body quickly.
 .
```
23/05/23 11:19:15,240 INFO [MemoryTracker-report-thread] MemoryTracker: Track all direct memory usage :26860.00000667572/40960.0,disk buffer size:112.91416263580322, sort memory size : 0.0
23/05/23 11:19:25,322 INFO [MemoryTracker-report-thread] MemoryTracker: Track all direct memory usage :26632.00000667572/40960.0,disk buffer size:111.59745025634766, sort memory size : 0.0
23/05/23 11:19:35,413 INFO [MemoryTracker-report-thread] MemoryTracker: Track all direct memory usage :26548.00000667572/40960.0,disk buffer size:116.29310417175293, sort memory size : 0.0
23/05/23 11:19:45,465 INFO [MemoryTracker-report-thread] MemoryTracker: Track all direct memory usage :26524.00000667572/40960.0,disk buffer size:573.5329294204712, sort memory size : 0.0
23/05/23 11:19:55,666 INFO [MemoryTracker-report-thread] MemoryTracker: Track all direct memory usage :26520.00000667572/40960.0,disk buffer size:1106.257067680359, sort memory size : 0.0
23/05/23 11:20:05,668 INFO [MemoryTracker-report-thread] MemoryTracker: Track all direct memory usage :27240.00000667572/40960.0,disk buffer size:1985.0912733078003, sort memory size : 0.0
23/05/23 11:20:15,708 INFO [MemoryTracker-report-thread] MemoryTracker: Track all direct memory usage :30780.00000667572/40960.0,disk buffer size:2212.156195640564, sort memory size : 0.0
23/05/23 11:20:25,711 INFO [MemoryTracker-report-thread] MemoryTracker: Track all direct memory usage :30780.00000667572/40960.0,disk buffer size:25.4876651763916, sort memory size : 0.0
23/05/23 11:20:35,715 INFO [MemoryTracker-report-thread] MemoryTracker: Track all direct memory usage :30320.00000667572/40960.0,disk buffer size:30.625568389892578, sort memory size : 0.0
23/05/23 11:20:46,218 INFO [MemoryTracker-report-thread] MemoryTracker: Track all direct memory usage :28416.00000667572/40960.0,disk buffer size:28.170005798339844, sort memory size : 0.0
23/05/23 11:20:56,225 INFO [MemoryTracker-report-thread] MemoryTracker: Track all direct memory usage :25836.00000667572/40960.0,disk buffer size:20.839608192443848, sort memory size : 0.0
23/05/23 11:21:06,225 INFO [MemoryTracker-report-thread] MemoryTracker: Track all direct memory usage :25732.00000667572/40960.0,disk buffer size:16.916643142700195, sort memory size : 0.0
23/05/23 11:21:16,225 INFO [MemoryTracker-report-thread] MemoryTracker: Track all direct memory usage :25660.00000667572/40960.0,disk buffer size:14.602804183959961, sort memory size : 0.0
23/05/23 11:21:26,227 INFO [MemoryTracker-report-thread] MemoryTracker: Track all direct memory usage :25608.00000667572/40960.0,disk buffer size:16.413521766662598, sort memory size : 0.0
23/05/23 11:21:36,230 INFO [MemoryTracker-report-thread] MemoryTracker: Track all direct memory usage :25592.00000667572/40960.0,disk buffer size:14.356255531311035, sort memory size : 0.0
23/05/23 11:21:46,232 INFO [MemoryTracker-report-thread] MemoryTracker: Track all direct memory usage :25564.00000667572/40960.0,disk buffer size:16.167262077331543, sort memory size : 0.0
23/05/23 11:21:56,233 INFO [MemoryTracker-report-thread] MemoryTracker: Track all direct memory usage :25564.00000667572/40960.0,disk buffer size:14.535367012023926, sort memory size : 0.0
23/05/23 11:22:06,233 INFO [MemoryTracker-report-thread] MemoryTracker: Track all direct memory usage :25556.00000667572/40960.0,disk buffer size:10.594958305358887, sort memory size : 0.0
23/05/23 11:22:16,408 INFO [MemoryTracker-report-thread] MemoryTracker: Track all direct memory usage :25556.00000667572/40960.0,disk buffer size:12.997883796691895, sort memory size : 0.0
23/05/23 11:22:26,412 INFO [MemoryTracker-report-thread] MemoryTracker: Track all direct memory usage :25556.00000667572/40960.0,disk buffer size:20.91703701019287, sort memory size : 0.0
23/05/23 11:22:36,412 INFO [MemoryTracker-report-thread] MemoryTracker: Track all direct memory usage :25556.00000667572/40960.0,disk buffer size:21.703519821166992, sort memory size : 16.0
23/05/23 11:22:46,412 INFO [MemoryTracker-report-thread] MemoryTracker: Track all direct memory usage :25556.00000667572/40960.0,disk buffer size:21.98106098175049, sort memory size : 16.0
23/05/23 11:22:56,412 INFO [MemoryTracker-report-thread] MemoryTracker: Track all direct memory usage :25556.00000667572/40960.0,disk buffer size:20.05210590362549, sort memory size : 16.0 
```

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @waitinfuture @FMX @RexXiong

File: common/src/main/java/org/apache/celeborn/common/network/protocol/ChunkFetchFailure.java [hunk 1]
```diff
@@ -49,7 +49,9 @@ public void encode(ByteBuf buf) {
   public static ChunkFetchFailure decode(ByteBuf buf) {
     StreamChunkSlice streamChunkSlice = StreamChunkSlice.decode(buf);
     String errorString = Encoders.Strings.decode(buf);
-    return new ChunkFetchFailure(streamChunkSlice, errorString);
+    ChunkFetchFailure chunkFetchFailure = new ChunkFetchFailure(streamChunkSlice, errorString);
+    chunkFetchFailure.setBody(buf);
```
Comment (FMX) [inline common/src/main/java/org/apache/celeborn/common/network/protocol/ChunkFetchFailure.java:53]: I think this change is not correct. Check the TransportFrameDecoder's logic below.
```
 public void channelRead(ChannelHandlerContext ctx, Object data) {
    ByteBuf in = (ByteBuf) data;
    buffers.add(in);
    totalSize += in.readableBytes();

    while (!buffers.isEmpty()) {
      ByteBuf frame = decodeNext();
      if (frame == null) {
        break;
      }
      Message msg = Message.decode(curType, frame);
      if (msg.body() == null) {
        frame.release();
      }
      ctx.fireChannelRead(msg);
      clear();
    }
  }
```

**Review by FMX — CHANGES_REQUESTED** [review]
I think this PR has some issues to figure out.

File: common/src/main/java/org/apache/celeborn/common/network/protocol/ChunkFetchFailure.java (hunk 1 above)
Comment (AngersZhuuuu) [reply common/src/main/java/org/apache/celeborn/common/network/protocol/ChunkFetchFailure.java:53]: Yea, will close this and next time meet this make a dump

**Pull request closed without merging**